app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordContents.java

```diff
@@ -0,0 +1,363 @@
+        // The keyboard uses the language layout of the user dictionary
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            mWordEditText.setImeHintLocales(new LocaleList(LocaleUtils.constructLocaleFromString(mLocale)));
+        }
+
+    }
+
+    void delete(final Context context) {
+        final ContentResolver resolver = context.getContentResolver();
+        final String localeInToast = new LocaleRenderer(context, mLocale).toString();
+        final String messageDeleted = context.getString(R.string.user_dict_word_deleted) + " " + localeInToast;
+        // Mode edit: remove the old entry.
+        if (MODE_EDIT == mMode && !TextUtils.isEmpty(mOldWord)) {
+            UserDictionarySettings.deleteWordInEditMode(mOldWord, mOldShortcut, mOldWeight, resolver);
+            // Toast appears to indicate that the word has been deleted
+            Toast.makeText(context, messageDeleted, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    public final int apply(@NonNull final Context context) {
+        final ContentResolver resolver = context.getContentResolver();
+        final String newWord = mWordEditText.getText().toString();
+        final String newShortcut;
+        final String newWeight;
+
+        if (null == mShortcutEditText) {
+            newShortcut = null;
+        } else {
+            final String tmpShortcut = mShortcutEditText.getText().toString();
+            if (TextUtils.isEmpty(tmpShortcut)) {
+                newShortcut = null;
+            } else {
+                newShortcut = tmpShortcut;
+            }
+        }
+
+        if (mWeightEditText == null) {
+            newWeight = String.valueOf(WEIGHT_FOR_USER_DICTIONARY_ADDS);
+        } else {
+            final String tmpWeight = mWeightEditText.getText().toString();
+            if (TextUtils.isEmpty(tmpWeight)) {
+                newWeight = String.valueOf(WEIGHT_FOR_USER_DICTIONARY_ADDS);
+            } else {
+                newWeight = tmpWeight;
+            }
+        }
+
+        if (TextUtils.isEmpty(newWord)) {
+            // If the word is empty, don't insert it.
+            return CODE_CANCEL;
+        }
+
+        mSavedWord = newWord;
+        mSavedShortcut = newShortcut;
+        mSavedWeight = newWeight;
+
+        // In edit mode, everything is modified without overwriting other existing words
+        if (MODE_EDIT == mMode && hasWord(newWord, context) && newWord.equals(mOldWord)) {
+            UserDictionarySettings.deleteWordInEditMode(mOldWord, mOldShortcut, mOldWeight, resolver);
+            // Toast appears to indicate that the word has been modified
+            Toast.makeText(context, context.getText(R.string.user_dict_word_modified), Toast.LENGTH_SHORT).show();
+        } else {
+            mMode = MODE_INSERT;
+        }
+
+        // If the word already exists, a dialog box appears asking you to change the word
+        // See UserDictionaryAddWordFragment.addWord()
```

Comment:
The comment is outdated, now the dialog doesn't ask anything.
Actually there is no need to specify / comment what happens. This should only be decided by the part receiving the return code (the code itself is already pretty self-explanatory)

Reply:
Change made with commit 0582df5